### PR TITLE
[Feature][Model] Support MiniMax M3 with CANN MegaMoE

### PR DESCRIPTION
**Pull request by CXY-Katrina**
### What this PR does / why we need it?

MiniMax M3 uses the `swigluoai_uninterleave` activation, so its MegaMoE path must forward the activation name together with the clamp, alpha, and beta parameters.

Assuming the latest `cann_ops_transformer` package provides `swigluoai` support, this PR keeps MiniMax M3 on the existing shared CANN MegaMoE implementation instead of introducing a local `_C_ascend.mega_moe` operator:

- removes the startup guard that rejected `enable_fused_mc2=1` for MiniMax M3;
- recognizes MiniMax M3 as MegaMoE-compatible by using its `intermediate_size` when `moe_intermediate_size` is absent;
- uses `enable_fused_mc2` as the single gate for fused-scale preparation, independent of activation type;
- retains the existing `_MEGA_MOE_SUPPORTED` check for selecting the CANN MegaMoE weight path;
- maps the internal activation name to `swigluoai` and passes `alpha`/`beta` through `activation_params`;
- splits inputs above the CANN MegaMoE BS limit of 4096 and writes chunk results into a preallocated output tensor to avoid `torch.cat` peak memory;
- adds regression coverage for config detection, weight selection, activation forwarding, chunking, output placement, masks, and expert-count accumulation.

This is a minimal alternative to #14096 and does not vendor the MegaMoE operator implementation.

### Does this PR introduce _any_ user-facing change?

Yes. MiniMax M3 can enable fused MC2 and use the shared CANN MegaMoE path when a `cann_ops_transformer` version with `swigluoai` support is installed. Token batches larger than 4096 are handled in chunks.

### How was this patch tested?

Added unit tests:

- `tests/ut/ops/test_cann_mega_moe.py`
- `tests/ut/quantization/methods/a2/test_w8a8_dynamic.py::TestAscendW8A8FusedMoEMethod::test_apply_swigluoai_uses_cann_mega_moe_weights`

Local checks completed:

```bash
python -m ruff format --check vllm_ascend/ascend_config.py vllm_ascend/ops/fused_moe/moe_comm_method.py vllm_ascend/quantization/methods/w8a8_dynamic.py tests/ut/ops/test_cann_mega_moe.py tests/ut/quantization/methods/a2/test_w8a8_dynamic.py
python -m ruff check vllm_ascend/ascend_config.py vllm_ascend/ops/fused_moe/moe_comm_method.py vllm_ascend/quantization/methods/w8a8_dynamic.py tests/ut/ops/test_cann_mega_moe.py tests/ut/quantization/methods/a2/test_w8a8_dynamic.py
python -m py_compile vllm_ascend/ascend_config.py vllm_ascend/ops/fused_moe/moe_comm_method.py vllm_ascend/quantization/methods/w8a8_dynamic.py tests/ut/ops/test_cann_mega_moe.py tests/ut/quantization/methods/a2/test_w8a8_dynamic.py
git diff --check
```

The unit tests could not be executed in the current Windows workspace because its Python environment does not contain `torch` or `pytest`, and no Ascend NPU/operator runtime is available. Per the requested scope, no service startup or NPU inference validation was performed.

- vLLM version: v0.27.1
- vLLM main: https://github.com/vllm-project/vllm/commit/ba07e4a48fc951300d97eb506217dd530583dea3

### CODE REVIEW

**Review by weijinqian0 — DISMISSED** [review]

**Comment by CXY-Katrina** [comment]
/cancel
[Bot]: successfully force-cancelled the following runs:


https://github.com/vllm-project/vllm-ascend/actions/runs/32686441537
